plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2961,6 +2997,80 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
+                // Statistics returns whole path to file build in DeltaLakeSplitManager, so we need to create corresponding map key for AddFileEntry.
+                .collect(toImmutableMap(addFileEntry -> DeltaLakeSplitManager.buildSplitPath(Location.of(tableHandle.getLocation()), addFileEntry).toString(), identity()));
+
+        if (addFileEntriesWithNoStats.isEmpty()) {
+            return;
+        }
+
+        Map<String, DeltaLakeColumnHandle> columnsHandles = getColumns(tableHandle.getMetadataEntry()).stream()
+                .filter(column -> column.getColumnType() == REGULAR)
+                .collect(toImmutableMap(columnHandle -> columnHandle.getBaseColumnName().toLowerCase(ENGLISH), identity()));
+
+        List<AddFileEntry> updatedAddFileEntries = computedStatistics.stream()
+                .map(statistics -> {
+                    // Grouping by `PATH_COLUMN_NAME`.
+                    String filePathFromStatistics = VARCHAR.getSlice(statistics.getGroupingValues().get(0), 0).toStringUtf8();
+                    // Check if collected statistics are for files without stats.
+                    // If AddFileEntry is present in addFileEntriesWithNoStats means that it does not have statistics so prepare updated entry.
+                    // If null is returned from addFileEntriesWithNoStats means that statistics are present, and we don't need to do anything.
+                    AddFileEntry addFileEntry = addFileEntriesWithNoStats.get(filePathFromStatistics);
+                    if (addFileEntry != null) {
+                        return Optional.of(prepareUpdatedAddFileEntry(statistics, addFileEntry, columnsHandles));
+                    }
+                    return Optional.<AddFileEntry>empty();
+                })
+                .flatMap(Optional::stream)
+                .collect(toImmutableList());
+
+        if (updatedAddFileEntries.isEmpty()) {
+            return;
+        }
+        try {
+            long createdTime = Instant.now().toEpochMilli();
+            long readVersion = tableHandle.getReadVersion();
+            long commitVersion = readVersion + 1;
+            TransactionLogWriter transactionLogWriter = transactionLogWriterFactory.newWriter(session, tableHandle.getLocation());
+            transactionLogWriter.appendCommitInfoEntry(getCommitInfoEntry(session, commitVersion, createdTime, OPTIMIZE_OPERATION, readVersion));
+            updatedAddFileEntries.forEach(transactionLogWriter::appendAddFileEntry);
+            transactionLogWriter.flush();
+        }
+        catch (Throwable e) {
+            throw new TrinoException(DELTA_LAKE_BAD_WRITE, "Unable to access file system for: " + tableHandle.getLocation(), e);
+        }
+    }
+
+    private AddFileEntry prepareUpdatedAddFileEntry(ComputedStatistics stats, AddFileEntry addFileEntry, Map<String, DeltaLakeColumnHandle> columnsHandles)
```

Comment:
Same for downstream methods. 
```suggestion
    private AddFileEntry prepareUpdatedAddFileEntry(ComputedStatistics stats, AddFileEntry addFileEntry, Map</* lowercase*/ String, DeltaLakeColumnHandle> lowercaseToColumnHandles)
```